integrations/utils/commonUtils.js

```diff
@@ -40,9 +39,12 @@ const getHashFromArrayWithDuplicate = (
         : array[fromKey].trim();
 
       if (hashMap[key]) {
-        hashMap[key].push(array[toKey]);
+        const val = hashMap[key];
+        if (!val.includes(toKey)) {
```

Comment:
What's the purpose of this check?

Reply:
We are checking if duplicate elements are added to the array or not. Let's say someone provided same event name mapping twice. Something like:
```
[
          {
            "from": "Downloaded",
            "to": "download"
          },
          {
            "from": "Downloaded",
            "to": "download"
          }
        ]
```
then the Downloaded key will contain the array with ["download", "download"]. To omit this situation we are doing this check.

Reply:
In that case, we should not be checking for the presence of `toKey` (which is the string literal `to`).

Reply:
Shouldn't it be `if (!val.includes(array[toKey])) {`?

Reply:
We decided to do that with help of set but that would be a breaking change.

Reply:
I don't understand. It has a logical error.

Reply:
Yes. Updated the code.